walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,206 @@
+      endTime = busyTime.start();
+      if (endTime - startTime >= eventDuration) {
+        availableTimes.add(TimeRange.fromStartEnd(startTime, endTime, false));
+      }
+
+      startTime = busyTime.end();
+    }
+
+    // Create the last availability window (if it fits the requested duration)
+    // from the end of the last event to the end of the day
+    if (TimeRange.END_OF_DAY - startTime >= eventDuration) {
+      availableTimes.add(TimeRange.fromStartEnd(startTime, TimeRange.END_OF_DAY, true));
+    }
+
+    return availableTimes;
+  }
+
+  /**
+   * Takes in a list of all events and merges them into non-overlapping time ranges of
+   * unavailability
+   *
+   * @param eventTimes a sorted list of event time ranges (from earliest start time to latest start
+   *     time)
+   * @return a sorted list of non-overlapping time ranges (from earliest start time to latest start
+   *     time)
+   */
+  private List<TimeRange> merge(List<TimeRange> eventTimes) {
+    // Create empty list to store busy periods
+    List<TimeRange> busyTimes = new ArrayList<>();
+
+    // Iterate through all of the events
+    // and merge overlapping events into one larger event
+    int currentEventIndex = 0;
+    while (currentEventIndex < eventTimes.size()) {
+      // The shortest the merged unavailable window can be is the length of the current window.
+      int startTime = eventTimes.get(currentEventIndex).start();
+      int endTime = eventTimes.get(currentEventIndex).end();
+
+      // Then, continue iterating through the list (if possible)
+      while (currentEventIndex < eventTimes.size() - 1) {
+        // This event is guaranteed to start later than the current startTime, since the
+        // event list is sorted
+        TimeRange compare = eventTimes.get(currentEventIndex + 1);
+
+        // Three Cases:
+        // Case 1: If the next event contains the end time of the previous,
+        // or starts when the previous one ends, the events are overlapping
+        // and the merged event should grow to the endTime of the second event
+        //
+        // Prev. Event(s) : |------|
+        // Next Event     :       |---|
+        // Merged Event   : |---------|
+        //
+        // Case 2: The next event is fully contained by the merged event
+        // Continue iterating and looking for more merges, but don't change current interval
+        // Prev. Event(s) : |-----------------|
+        // Next Event     :      |--|
+        // Merged Event   : |-----------------|
+        //
+        // Case 3: If the next event starts later than the previous one ends,
+        // a gap of availability is identified. Merging can now stop
+        //
+        // Prev. Event(s) : |------|
+        // Next Event     :          |-----|
+        // No merge. Move on.
+        //
+        if (compare.contains(endTime) || compare.start() == endTime) {
+          endTime = eventTimes.get(currentEventIndex + 1).end();
```

Comment:
Nit: Use the `compare` variable you defined

Reply:
Oops - fixed